optuna/_gp/search_space.py

```diff
@@ -75,7 +75,11 @@ def round_one_normalized_param(
         return param_value
 
     param_value = unnormalize_one_param(param_value, scale_type, bounds, step)
-    param_value = (param_value - bounds[0] + 0.5 * step) // step * step + bounds[0]
+    param_value = np.clip(
+        (param_value - bounds[0] + 0.5 * step) // step * step + bounds[0],
+        bounds[0],
+        bounds[1],
+    )
```

Comment:
```suggestion
    n_grids = np.round((bounds[1] - bounds[0] + step) / step)
    grid_indices = (param_value - bounds[0] + 0.5 * step) // step
    # NOTE(bug): https://github.com/optuna/optuna/pull/5543/
    param_value = np.clip(grid_indices, 0, n_grids - 1) * step + bounds[0]
```

Reply:
This might return invalid value like 1.00000000....1 by numerical error.